source/core/src/main/com/csse3200/game/components/items/ItemActions.java

```diff
@@ -308,9 +308,12 @@ private boolean hoe(Vector2 playerPos, Vector2 mousePos) {
    */
   private boolean fertilise(TerrainTile tile) {
     if (isCropTile(tile.getOccupant())) {
+      CropTileComponent cropTileComponent = tile.getOccupant().getComponent(CropTileComponent.class);
+      if (cropTileComponent != null && !cropTileComponent.isFertilised()) {
```

Comment:
if cropTile is null this will be like NoneType has no isFertilised() method

Reply:
but isCropTile() checks that so wont actually happen just checked twice ig

Reply:
That's why I check that it's not null in the previous condition of the && statement

Reply:
It shortcircuits if cropTileComponent is null

Reply:
also isCropTile is also checked immediately beforehand

Reply:
cropTile checks if it has a CropTileComp so the first half of ur if is useless, and and if statement checks both sides before returning so would still break

Reply:
Would the second if statement not just short circuit? If the component was null, the first condition would evaluate to `false` then it wouldn't bother evaluating `!cropTileComponent.isFertilised()`

Reply:
I've removed the double-check for cropTileComponent being not null, but && is shortcircuit so it wouldn't